app/javascript/react/components/NewMovieForm.js

```diff
@@ -0,0 +1,59 @@
+import React, {useState} from "react"
+import NewMovieFormTile from "./NewMovieFormTile"
+import { Redirect } from "react-router-dom"
+
+const NewMovieForm = (props) => {
+
+  const [success, setSuccess] = useState({movieSaved: false, response: []})
+
+  const addMovie = async (formPayload) => {
+    try {
+      const response = await fetch("/api/v1/movies", {
+        credentials: "same-origin",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json"
+        },
+        body: JSON.stringify(formPayload)
+      })
+      if(!response.ok){
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const responseBody = await response.json()
+      setSuccess(responseBody)
```

Comment:
here we would check if `responseBody.movie` exists - if it does, we set state to redirect. If `responseBody.errors` then we can set a state for errors that will display the error messages